src/drivers/intel/ssp/ssp.c

```diff
@@ -766,19 +780,80 @@ static int ssp_get_hw_params(struct dai *dai,
 	return 0;
 }
 
-/* SSP hw params */
+/*
+ * enabled clock MCLK/BCLK to support codec that need MCLK/BCLK
+ * This is called before trigger(COMP_TRIGGER_START)
+ */
 static int ssp_hw_params(struct dai *dai,
 			 struct sof_ipc_stream_params  *params)
 {
-	dai_info(dai, "ssp_hw_params()");
+	struct ssp_pdata *ssp = dai_get_drvdata(dai);
```

Comment:
have you checked that the hw_params is called after set_config? if not, the values will be uninitialized.